sdk/core/core/src/az_http_policy.c

```diff
@@ -61,12 +65,110 @@ AZ_NODISCARD az_result az_http_pipeline_policy_retry(
     void * const data,
     az_http_request_builder * const hrb,
     az_http_response * const response) {
-  (void)data;
-
   // reset response to be written from the start
-  AZ_RETURN_IF_FAILED(az_http_response_reset(response));
+  if (data == NULL) {
+    AZ_RETURN_IF_FAILED(az_http_response_reset(response));
+    return az_http_pipeline_nextpolicy(p_policies, hrb, response);
+  }
 
-  return az_http_pipeline_nextpolicy(p_policies, hrb, response);
+  az_http_policy_retry_options const * const retry_policy
+      = (az_http_policy_retry_options const *)data;
+
+  int16_t const max_tries = retry_policy->max_tries;
+
+  bool const custom_retry_delay_limits = retry_policy->retry_delay_msec > 0
```

Comment:
It would be better to validate this when the options are passed to initialize a client and then fail client creation instead of silently ignoring the options while requests are under way and using values the user did NOT specify.